ptxcompiler/patch.py

```diff
@@ -107,6 +107,37 @@ def get_cubin(self, cc=None):
 def patch_needed():
     logger = get_logger()
 
+    # The patch is needed if the user explicitly forced it with an environment
+    # variable.
+    apply = os.getenv("PTXCOMPILER_APPLY_NUMBA_CODEGEN_PATCH")
+    if apply is not None:
+        logger.debug(f"PTXCOMPILER_APPLY_NUMBA_CODEGEN_PATCH={apply}")
+        try:
+            apply = int(apply)
+        except ValueError:
+            apply = False
+
+    if apply:
+        return True
+
+    # We should avoid checking whether the patch is needed if the user
+    # requested that we don't check (e.g. in a non-fork-safe environment)
+    check = os.getenv("PTXCOMPILER_CHECK_NUMBA_CODEGEN_PATCH_NEEDED")
+    if check is not None:
+        logger.debug(f"PTXCOMPILER_CHECK_NUMBA_CODEGEN_PATCH_NEEDED={check}")
+        try:
+            check = int(check)
+        except ValueError:
+            check = False
+    else:
+        check = True
+
+    if not check:
+        return False
```

Comment:
```suggestion
    if check is not None:
        logger.debug(f"PTXCOMPILER_CHECK_NUMBA_CODEGEN_PATCH_NEEDED={check}")
        try:
            return int(check)
        except ValueError:
            return False
```

Reply:
If the user sets `PTXCOMPILER_CHECK_NUMBA_CODEGEN_PATCH_NEEDED` to `1`, won't this end up always applying the patch instead of checking if the patch is needed?

Reply:
Ah sorry, yes.